dotnet/src/Connectors/Connectors.OpenAIV2/Core/ClientCore .TextToAudio.cs

```diff
@@ -0,0 +1,76 @@
+- Added Checking for empty or whitespace prompt.
+- Removed the format parameter as this is never called in V1 code. Plan to implement it in the future once we change the ITextToImageService abstraction, using PromptExecutionSettings.
+- Allow custom size for images when the endpoint is not the default OpenAI v1 endpoint.
+*/
+
+using System;
+using System.ClientModel;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenAI.Audio;
+
+namespace Microsoft.SemanticKernel.Connectors.OpenAI;
+
+/// <summary>
+/// Base class for AI clients that provides common functionality for interacting with OpenAI services.
+/// </summary>
+internal partial class ClientCore
+{
+    /// <summary>
+    /// Generates an image with the provided configuration.
+    /// </summary>
+    /// <param name="prompt">Prompt to generate the image</param>
+    /// <param name="executionSettings">Text to Audio execution settings for the prompt</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Url of the generated image</returns>
+    internal async Task<IReadOnlyList<AudioContent>> GetAudioContentsAsync(
+        string prompt,
+        PromptExecutionSettings? executionSettings,
+        CancellationToken cancellationToken)
+    {
+        Verify.NotNullOrWhiteSpace(prompt);
+
+        OpenAITextToAudioExecutionSettings? audioExecutionSettings = OpenAITextToAudioExecutionSettings.FromExecutionSettings(executionSettings);
+        var (responseFormat, mimeType) = GetGeneratedSpeechFormatAndMimeType(audioExecutionSettings?.ResponseFormat);
+        SpeechGenerationOptions options = new()
+        {
+            ResponseFormat = responseFormat,
+            Speed = audioExecutionSettings?.Speed,
+        };
+
+        ClientResult<BinaryData> response = await RunRequestAsync(() => this.Client.GetAudioClient(this.ModelId).GenerateSpeechFromTextAsync(prompt, GetGeneratedSpeechVoice(audioExecutionSettings?.Voice), options, cancellationToken)).ConfigureAwait(false);
+
+        return [new AudioContent(response.Value.ToArray(), mimeType)];
+    }
+
+    private static GeneratedSpeechVoice GetGeneratedSpeechVoice(string? voice)
+        => voice?.ToUpperInvariant() switch
+        {
+            "ALLOY" => GeneratedSpeechVoice.Alloy,
+            "ECHO" => GeneratedSpeechVoice.Echo,
+            "FABLE" => GeneratedSpeechVoice.Fable,
+            "ONYX" => GeneratedSpeechVoice.Onyx,
+            "NOVA" => GeneratedSpeechVoice.Nova,
+            "SHIMMER" => GeneratedSpeechVoice.Shimmer,
+            _ => throw new NotSupportedException($"The voice '{voice}' is not supported."),
+        };
+
+    private static (GeneratedSpeechFormat, string) GetGeneratedSpeechFormatAndMimeType(string? format)
```

Comment:
Consider using new private struct to be more clear instead of valuetuple.

Reply:
Will consider when the method is used in more places, for now this is fine, added named tuples to make it more legible from the Method definition.